python/scannerpy/column.py

```diff
@@ -233,22 +233,16 @@ def load(self, ty=None, fn=None, rows=None, workers=16):
                 protobufs.VideoDescriptor.H264):
             png_table_name = self._db._png_dump_prefix.format(
                 self._table.name(), self._name)
-            pair = [(self._table.name(), png_table_name)]
-            op_args = {}
-            frame = self._db.sources.FrameColumn()
-            op_args[frame] = self
+            frame = self._db.io.Input([ScannerFrameStream(self._db, self._table.name())])
             enc_input = frame
             if rows is not None:
-                sampled_frame = self._db.streams.Gather(frame, rows=rows)
+                sampled_frame = self._db.streams.Gather(frame, indices=[rows])
                 enc_input = sampled_frame
             img = self._db.ops.ImageEncoder(frame=enc_input)
-            output_op = self._db.sinks.Column(columns={'img': img})
-            op_args[output_op] = png_table_name
-            job = Job(op_args=op_args)
-            self._db.run(
-                output_op, [job], force=True, show_progress=False)
-            out_tbl = self._db.table(png_table_name)
-            return out_tbl.column('img').load(fn=readers.image)
+            output = [ScannerStream(self._db, png_table_name)]
```

Comment:
Can we file an issue to avoid doing this? It's really gross that we explode the video into images to return them to the user.

Reply:
Yeah sure.